### PR TITLE
removeDuplicate query optimized to partially fix #799

### LINKED ISSUE #799: Event db. Improve remove duplicates.
We want to avoid acting on the same event twice. The current method, [removeDuplicate](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/event.go#L75), is inefficient as it queries all events. 

Especially in conjunction with https://github.com/0chain/0chain/issues/756 we can assume that blocks will not be repeated. So we should only need to search events for one block for repeats.

### PR DESCRIPTION
**Pull request by samtholiya**
Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [x]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a branch at [0chain/system_test](https://github.com/0chain/system_test) then add and merge your test changes to master BEFORE merging this PR

### CODE REVIEW

**Comment by samtholiya** [comment]
```
zchain_unit_test_1  | ?         0chain.net/smartcontract/benchmark/main/cmd/log [no test files]
zchain_unit_test_1  | ?         0chain.net/smartcontract/dbs    [no test files]
zchain_unit_test_1  | ok        0chain.net/smartcontract/dbs/event      0.480s  coverage: 29.4% of statements
zchain_unit_test_1  | ?         0chain.net/smartcontract/dbs/mocket     [no test files]
zchain_unit_test_1  | ?         0chain.net/smartcontract/dbs/postgresql [no test files]
zchain_unit_test_1  | ?         0chain.net/smartcontract/faucetsc       [no test files]
```

Event query passed

**Review by kushthedude — COMMENTED** [review]
The query is faster but I dont see if there would be any changes in the result expected of it 🤔 ?

**Review by Sriep — APPROVED** [review]
.

**Review by Sriep — CHANGES_REQUESTED** [review]
This still searches a potentially large table looking for a duplicate. Has this been benchmarked for extremely large events tables? I am not convinced this will solve the reported problem of the event database taking too long on mature chains.

Can't we just keep a local list of events from the current block and search that? Events are received in order https://github.com/0chain/0chain/issues/757, so if an event for another block is processed we can drop the list for the previously current block and start with the new block.

This way we only search hundreds of events from one block rather than millions or billions of events across the chain's history.

File: code/go/0chain.net/smartcontract/dbs/event/event.go [hunk 1]
```diff
@@ -61,16 +62,16 @@ func (edb *EventDb) GetEvents(ctx context.Context, block int64) ([]Event, error)
 }
 
 func (edb *EventDb) exists(ctx context.Context, event Event) (bool, error) {
-	var count int64
-	result := edb.Store.Get().WithContext(ctx).
-		Model(&Event{}).
-		Where("tx_hash = ? AND index = ?", event.TxHash, event.Index).
-		Count(&count)
+	var ev Event
+	result := edb.Store.Get().WithContext(ctx).Model(&Event{}).Where(&Event{TxHash: event.TxHash, Index: event.Index}).Take(&ev)
```
Comment (ssardana96) [inline code/go/0chain.net/smartcontract/dbs/event/event.go]: shouldn't we be checking type and tag here as well @Sriep

**Review by ssardana96 — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/event.go (hunk 1 above)
Comment (samtholiya) [reply code/go/0chain.net/smartcontract/dbs/event/event.go]: done

**Comment by peterlimg** [comment]
`removeDuplicate` should not be needed once https://github.com/0chain/0chain/issues/993 is done.

**Comment by samtholiya** [comment]
as @peterlimg told this pr is not needed therefore closing this pr

**Pull request closed without merging**